docs/excel/custom-functions-architecture.md

```diff
@@ -0,0 +1,42 @@
+---
+ms.date: 03/06/2019
+description: Learn about Excel custom functions' runtime. 
+title: Custom functions architecture (preview)
+localization_priority: Priority
+---
+# Custom functions architecture
+
+Custom function use a new JavaScript runtime that differs from the browser-based JavaScript engine which powers most other parts of your add-in.
+
+## Custom functions runtime
+
+The custom functions runtime prioritizes executing calculations, so functions run run smoothly when Excel recalculates. The custom functions runtime allows access to APIs which make requesting external data and exchanging data over a persistent connection over a server possible.
+
+Because functions are limited to actions performed in a workbook, the custom functions runtime doesn't support UI elements nor does it provide access to the Document Object Model (DOM). You should think of custom functions as a separate component of your add-in, separate from your add-in's task pane or other UI elements.
```

Comment:
"limited to actions performed in a workbook" sort of implies you can have UI since task panes also perform actions in a workbook like UI interactions. Could be clearer to say that custom functions purpose is to return a calculated value for a cell and nothing else.

Reply:
I wonder if we should have a bullet list of these items. At any rate, we should be more clear about what it means to not have a DOM. for example, you cannot use JQuery or any other libraries that would rely on a DOM.

Also we should mention that you cannot use local storage to share data with task panes or other sections of code. Each runtime has its own local storage. (Maybe this would be good to mention in AsyncStorage later just to help prevent developers from accidentally thinking they can use local storage.)